src/ispc.cpp

```diff
@@ -1016,7 +1016,7 @@ Target::Target(Arch arch, const char *cpu, ISPCTarget ispc_target, bool pic, boo
         this->m_dataTypeWidth = 8;
         this->m_vectorWidth = 16;
         this->m_hasHalf = true; // ??
-        this->m_maskingIsFree = false;
```

Comment:
Why is this change? NEON doesn't have mask registers, so masking is not "free".

Reply:
You are right.

`m_maskingIsFree` is used in https://github.com/ispc/ispc/blob/main/src/func.cpp#L340-L350
which is an optimization under which ispc creates a branch for `all_on` and `some_on`, and simplification of the branch condition sometimes leads to code without mask.

For AArch64 simplifycfg is not able to remove the branch as the back-end thinks that it is more beneficial to form FMAs in each branch: https://reviews.llvm.org/D7299 
The branch is removed once the back-end was able to form the FMA, however that happens very late and by that time DCE does not run anymore to remove the `tst` block of code that looks useless in bug #2052.

Reply:
Thanks for the explanation. This sounds like reasonable motivation to have this. I've been looking at this all_on/all_off optimization recently and thinking if it worth keeping. This seems to add more problems than have benefits. This is one more reasons to disable it by default.

If you'd like to keep it - add it as a separate commit and add comment that it's workaround for this problem.

Reply:
I opened a separate PR #2064 to disable this optimization on Arm64.